packages/react-components/react-tags/src/components/Tag/Tag.types.ts

```diff
@@ -1,20 +1,46 @@
+import { AvatarSize, AvatarShape } from '@fluentui/react-avatar';
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
-import { Avatar } from '@fluentui/react-avatar';
+import { Button } from '@fluentui/react-button';
+
+export type TagContextValues = {
+  avatar: {
+    size?: AvatarSize;
+    shape?: AvatarShape;
+  };
+};
 
 export type TagSlots = {
   root: NonNullable<Slot<'div'>>;
-  content?: Slot<'span'>;
-  avatar?: Slot<typeof Avatar>;
-  icon?: Slot<'span'>;
-  primaryText?: Slot<'span'>;
-  secondaryText?: Slot<'span'>;
-  dismissButton?: NonNullable<Slot<'button'>>;
+
+  /**
+   * Slot for an icon or other visual element
+   */
+  media: Slot<'span'>;
+
+  /**
+   * A layout wrapper for the icon slot, the primaryText and secondaryText slots
+   */
+  content: Slot<'div'>;
+
+  icon: Slot<'span'>;
+
+  /**
+   * Main text for the Tag. Children of the root slot are automatically rendered here
+   */
+  primaryText: Slot<'span'>;
+
+  /**
+   * Secondary text that describes or complements the main text
+   */
+  secondaryText: Slot<'span'>;
+
+  dismissButton: Slot<typeof Button>;
```

Comment:
One of the reasons to eliminate the `Avatar` slot in favor of media is to reduce the overall bundle size of `Tag`. How do we benefit by using Fluent `Button` here as it will also increase the bundle size?

Reply:
Good point :) I think there's no reason to use Button here. I updated it to use `Slot<'button'>`.